sdks/python/apache_beam/runners/portability/portable_stager_test.py

```diff
@@ -0,0 +1,148 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Test cases for :module:`artifact_service_client`."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import filecmp
+import logging
+import os
+import random
+import shutil
+import string
+import tempfile
+import unittest
+from concurrent import futures
+
+import grpc
+
+from apache_beam.portability.api import beam_artifact_api_pb2
+from apache_beam.portability.api import beam_artifact_api_pb2_grpc
+from apache_beam.runners.portability import portable_stager
+
+
+class PortableStagerTest(unittest.TestCase):
+
+  def setUp(self):
+    self._temp_dir = tempfile.mkdtemp()
+    self._remote_dir = tempfile.mkdtemp()
+
+  def tearDown(self):
+    if self._temp_dir:
+      shutil.rmtree(self._temp_dir)
+    if self._remote_dir:
+      shutil.rmtree(self._remote_dir)
+
+  def stage_files(self, files):
+    server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
+    beam_artifact_api_pb2_grpc.add_ArtifactStagingServiceServicer_to_server(
+        TestLocalFileSystemArtifactStagingServiceServicer(self._remote_dir),
+        server)
+    test_port = server.add_insecure_port('[::]:0')
+    server.start()
+    stager = portable_stager.PortableStager(
+        grpc.insecure_channel('localhost:%s' % test_port))
+    for from_file, to_file in files:
+      stager.stage_artifact(
+          local_path_to_artifact=os.path.join(self._temp_dir, from_file),
+          artifact_name=to_file)
+
+    return stager._artifacts
+
+  def test_stage_single_file(self):
+    from_file = 'test_local.txt'
+    to_file = 'test_remote.txt'
+
+    with open(os.path.join(self._temp_dir, from_file), 'wb') as f:
+      f.write(b'abc')
+
+    copied_files = self.stage_files([('test_local.txt', 'test_remote.txt')])
+    self.assertTrue(
+        filecmp.cmp(
+            os.path.join(self._temp_dir, from_file),
+            os.path.join(self._remote_dir, to_file)))
+    self.assertEqual([to_file], [manifest.name for manifest in copied_files])
```

Comment:
Manifest is a **list** of staged files, so it's not the best name for the loop variable here. If we want to check that manifest is created and staged, I would actually exercise commit_manifest behavior.

Reply:
Renamed to staged_file_metadata.

as mentioned above, commit_manifest is exercised automatically.